data-pipeline/src/trace_exporter/mod.rs

```diff
@@ -631,26 +635,38 @@ impl TraceExporter {
 
         match self.output_format {
             TraceExporterOutputFormat::V04 => {
-                let tracer_payload = trace_utils::collect_trace_chunks(
-                    TraceCollection::V04(traces),
-                    &header_tags,
-                    &mut tracer_payload::DefaultTraceChunkProcessor,
-                    self.endpoint.api_key.is_some(),
-                );
+                let payload = rmp_serde::to_vec_named(&traces).map_err(TraceExporterError::from)?;
+                let chunks = traces.len();
+                let payload_len = payload.len();
                 let endpoint = Endpoint {
                     url: self.output_format.add_path(&self.endpoint.url),
                     ..self.endpoint.clone()
                 };
-                let send_data = SendData::new(size, tracer_payload, header_tags, &endpoint);
+                let mut headers: HashMap<&str, String> = header_tags.into();
+                headers.insert(DATADOG_SEND_REAL_HTTP_STATUS_STR, "1".to_string());
+                headers.insert(DATADOG_TRACE_COUNT_STR, chunks.to_string());
+                headers.insert(CONTENT_TYPE.as_str(), APPLICATION_MSGPACK_STR.to_string());
+
+                let strategy = RetryStrategy::default();
                 self.runtime.block_on(async {
-                    let send_data_result = send_data.send().await;
+                    // Send traces to the agent
+                    let result =
+                        send_with_retry(&endpoint, payload, &headers, &strategy, None).await;
+
+                    // Send telemetry for the payload sending
                     if let Some(telemetry) = &self.telemetry {
-                        if let Err(e) = telemetry.send(&send_data_result) {
+                        if let Err(e) = telemetry.send(&SendPayloadTelemetry::from_retry_result(
+                            &result,
+                            payload_len.try_into().unwrap_or(0),
+                            chunks.try_into().unwrap_or(0),
```

Comment:
nit: here you can use `as u64`, in this situation is safe to perform an unchecked casting since usize is always 32 or 64 bit depending on the architecture. Besides it'll probably be slightly faster due `try_into` + `unwrap` could result in extra branching but I haven't disassemble the code to check it.